samples/tfx/taxi-cab-classification-pipeline-on-prem.py

```diff
@@ -0,0 +1,200 @@
+                                 project, analyze_mode, analyze_slice_column, analysis_output,
+                                 step_name='analysis'):
+    return dsl.ContainerOp(
+        name=step_name,
+        image='gcr.io/ml-pipeline/ml-pipeline-dataflow-tfma:5df2cdc1ed145320204e8bc73b59cdbd7b3da28f',
+        arguments=[
+            '--model', model,
+            '--eval', evaluation_data,
+            '--schema', schema,
+            '--project', project,
+            '--mode', analyze_mode,
+            '--slice-columns', analyze_slice_column,
+            '--output', '%s/{{workflow.name}}/analysis' % analysis_output,
+        ],
+        file_outputs={'analysis': '/output.txt'}
+    )
+
+
+def dataflow_tf_predict_op(evaluation_data, schema, target: str,
+                           model: 'TensorFlow model', predict_mode, project, prediction_output,
+                           step_name='prediction'):
+    return dsl.ContainerOp(
+        name=step_name,
+        image='gcr.io/ml-pipeline/ml-pipeline-dataflow-tf-predict:5df2cdc1ed145320204e8bc73b59cdbd7b3da28f',
+        arguments=[
+            '--data', evaluation_data,
+            '--schema', schema,
+            '--target', target,
+            '--model', model,
+            '--mode', predict_mode,
+            '--project', project,
+            '--output', '%s/{{workflow.name}}/predict' % prediction_output,
+        ],
+        file_outputs={'prediction': '/output.txt'}
+    )
+
+
+def confusion_matrix_op(predictions, output, step_name='confusion_matrix'):
+    return dsl.ContainerOp(
+        name=step_name,
+        image='gcr.io/ml-pipeline/ml-pipeline-local-confusion-matrix:5df2cdc1ed145320204e8bc73b59cdbd7b3da28f',
+        arguments=[
+            '--output', '%s/{{workflow.name}}/confusionmatrix' % output,
+            '--predictions', predictions,
+            '--target_lambda', """lambda x: (x['target'] > x['fare'] * 0.2)""",
+        ])
+
+
+def roc_op(predictions, output, step_name='roc'):
+    return dsl.ContainerOp(
+        name=step_name,
+        image='gcr.io/ml-pipeline/ml-pipeline-local-roc:5df2cdc1ed145320204e8bc73b59cdbd7b3da28f',
+        arguments=[
+            '--output', '%s/{{workflow.name}}/roc' % output,
+            '--predictions', predictions,
+            '--target_lambda', """lambda x: 1 if (x['target'] > x['fare'] * 0.2) else 0""",
+        ])
+
+
+def kubeflow_deploy_op(model: 'TensorFlow model', tf_server_name, pvc_name, step_name='deploy'):
+    return dsl.ContainerOp(
+        name=step_name,
+        image='gcr.io/ml-pipeline/ml-pipeline-kubeflow-deployer:5df2cdc1ed145320204e8bc73b59cdbd7b3da28f',
+        arguments=[
+            '--cluster-name', 'tfx-taxi-pipeline-on-prem',
+            '--model-path', model,
+            '--server-name', tf_server_name,
+            '--pvc-name', pvc_name,
+        ]
+    )
+
+
+@dsl.pipeline(
+    name='TFX Taxi Cab Classification Pipeline Example',
```

Comment:
Shall we add `on-prem` for name and description here? Also does the pipeline gcp file also need some update for name and description?